Dockerfile

```diff
@@ -1,24 +1,16 @@
-ARG PYTHON_VERSION=3.14
-
-FROM python:${PYTHON_VERSION}-slim AS base
-
-# Any python libraries that require system libraries to be installed will likely
-# need the following packages in order to build
-RUN apt-get update && \
-    apt-get -y upgrade && \
-    apt-get install -y build-essential git && \
-    apt-get clean && \
-    rm -rf /var/lib/apt/lists/*
-
-ENV CURL_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt
-
-FROM base AS builder
+FROM ghcr.io/astral-sh/uv:python3.14-bookworm-slim
 
 WORKDIR /app
 
+RUN apt-get update && apt-get install -y gcc
 COPY . /app
 
-RUN python -m pip install \
-    -e ./stac_fastapi/types \
-    -e ./stac_fastapi/api \
-    -e ./stac_fastapi/extensions
+ENV CURL_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt
+ENV UV_COMPILE_BYTECODE=1
+ENV UV_LINK_MODE=copy
+ENV UV_TOOL_BIN_DIR=/usr/local/bin
+RUN --mount=type=cache,target=/root/.cache/uv \
+  --mount=type=bind,source=uv.lock,target=uv.lock \
+  uv sync --locked --no-dev
```

Comment:
I'm not sure to follow where and when the package will be installed 

The `uv.lock` was created when doing `uv sync --dev` which contains a lot of dependencies not needed within the container which is why I kept the `python -m pip install`